sdks/python/apache_beam/dataframe/expressions.py

```diff
@@ -24,60 +30,77 @@ class Session(object):
   def __init__(self, bindings={}):
     self._bindings = dict(bindings)
 
-  def evaluate(self, expr):
+  def evaluate(self, expr):  # type: (Session, Expression) -> Any
     if expr not in self._bindings:
       self._bindings[expr] = expr.evaluate_at(self)
     return self._bindings[expr]
 
-  def lookup(self, expr):
+  def lookup(self, expr):  #  type: (Session, Expression) -> Any
     return self._bindings[expr]
 
 
+# The return type of an Expression
+T = TypeVar('T')
+
+
 class Expression(object):
   """An expression is an operation bound to a set of arguments.
 
   An expression represents a deferred tree of operations, which can be
   evaluated at a specific bindings of root expressions to values.
   """
-  def __init__(self, name, proxy=None, _id=None):
+  def __init__(
+      self,
+      name,  # type: str
+      proxy,  # type: T
+      _id=None  # type: Optional[str]
+  ):
     self._name = name
     self._proxy = proxy
     # Store for preservation through pickling.
     self._id = _id or '%s_%s' % (name, id(self))
 
-  def proxy(self):
+  def proxy(self):  # type: (Expression) -> T
     return self._proxy
 
   def __hash__(self):
     return hash(self._id)
 
-  def __eq__(self):
+  def __eq__(self, other):
```

Comment:
Oops...

Reply:
Haha yeah. Thanks to mypy for finding this one :)